sklearn/utils/_array_api.py

```diff
@@ -10,7 +10,7 @@
 from .._config import get_config
 from .fixes import parse_version
 
-_NUMPY_NAMESPACE_NAMES = {"numpy", "array_api_compat.numpy", "numpy.array_api"}
+_NUMPY_NAMESPACE_NAMES = {"numpy", "array_api_compat.numpy"}
```

Comment:
I support this change, but not including `array_api_strict` here is a change from how we have been treating `numpy.array_api`. It might make some uses of `is_numpy_namespace()` obsolete.

Related: do we need to keep `numpy.array_api` here for people who have an older numpy version and keep using it and expect that support for something experimental continues to exist in scikit-learn? I think we shouldn't keep it in and just assume that there are ~0 people in the world who use `numpy.array_api`.

Reply:
> It might make some uses of `is_numpy_namespace()` obsolete.

Note necessarily: it's useful to detect cases where:

- `sklearn.get_config(array_api_dispatch=True)` and the input is a (wrapped) numpy array.

vs

- `sklearn.get_config(array_api_dispatch=True)` and the input is not a (wrapped) numpy array.

This is important to know when we can safely convert back and forth to numpy without overhead, for instance to use the `out=` kwarg for memory efficiency reasons.


> do we need to keep numpy.array_api here for people who have an older numpy version and keep using it and expect that support for something experimental continues to exist in scikit-learn?

I don't see any value in this. `numpy.array_api` was a temporary experiment and it's going away. There is no point in using it in the future: it just adds complexity for no benefit.

`numpy.array_api`'s value was mostly for testing array api compliance, and this is now better served by `array-api-strict` and it works even with older numpy versions as demonstrated by our CI.